Adds Navigation V2 menu

tiffanybbrown-va opened the pull request:
Closes department-of-veterans-affairs/vets.gov-team#881

Commit:
Navigation level 2 layout complete.

- Removed CSS that's no longer in use.
- Adjusted markup of header to accommodate new nav.

Commit:
Added DropDown user account menu

- Added DropDown.jsx common component.
- Added SVG user icon component.

Work on department-of-veterans-affairs/vets.gov-team#881

Commit:
Added SearchMenu, HelpMenu, IconHelp components.

Work on department-of-veterans-affairs/vets.gov-team#881

Commit:
Added SearchMenu, HelpMenu, IconHelp components.

Work on department-of-veterans-affairs/vets.gov-team#881

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

Commit:
Merge branch 'department-of-veterans-affairs/vets.gov-team-881' of github.com-tiffanybbrown-va:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/vets.gov-team-881

melwoodard reviewed (dismissed):
@tiffanybbrown-va this looks great, and I just realized I never gave you proper design specs! Just a couple things:

1. I can open both the search and help menus at the same time, which isn't ideal. Can they behave the same way as the explore / manage menus, where a click on one will trigger a close of the other?
2. Per the spec above, can opening the search menu automatically focus the cursor into the search field, to save me the extra click?
3. `Sign in | Register` bleeds out of view on mobile (320px). Is hiding the search/help icons at the small breakpoint feasible re: [this comment](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/881#issuecomment-283713920) to help it fit?

<img width="320" src="https://cloud.githubusercontent.com/assets/4106356/23567338/b6520338-000a-11e7-874f-14aa2e0943b3.png">

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

Commit:
Hide button icons on smaller viewports.

tiffanybbrown-va commented:
Hiding the icons works!

![screen shot 2017-03-06 at 18 03 42-fullpage](https://cloud.githubusercontent.com/assets/25233690/23638582/4e48f7b2-0297-11e7-9a56-65c4a2083d03.png)

Commit:
Rewriting media queries in va.scss, disability-benefits.scss

Related to department-of-veterans-affairs/vets.gov-team#818

- Removed @media {#small} queries; these compiled to `@media only screen`
- Rewrote disability-benefits.scss media queries
- Rewrote medium-screen media queries in va.scss to use USWDS @include media

melwoodard commented:
:100: Awesome! Let me know when you're ready for another review.

annekainicUSDS committed:
Initial page setup

Add more fields

Write tests

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

tiffanybbrown-va commented:
@melwoodard Ready for another review.

Commit:
Fixed broken media query.

melwoodard commented:
@tiffanybbrown-va the search/sign in looks much better. Where are you at with 1 and 2 above?

tiffanybbrown-va commented:
@melwoodard Oops. Totally missed 1 & 2.

Commit:
Adding focus to search input when visible.

Commit:
Rewrote or removed Foundation media queries.

Related to:

  - department-of-veterans-affairs/vets.gov-team#818
  - department-of-veterans-affairs/vets.gov-team#638.

Changes:

- Rewrote Foundation-based media query in form elements partial.
- Rewrote Foundation media query for m-table.
- Removed Foundation @media {small} media queries.
    - Compiled to @media screen only, making them unnecessary.
- Rewrote media queries for cards module.
- Adjusted spacing above, below home page cards.
- Rewrote Foundation-based queries for modal.

annekainicUSDS committed:
Update test and change labels

annekainicUSDS committed:
Make review updates

cvalarida committed:
Update OMB (#5061)

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

tiffanybbrown-va commented:
> I can open both the search and help menus at the same time, which isn't ideal. Can they behave the same way as the explore / manage menus, where a click on one will trigger a close of the other?

This will be a heavier lift because it involves adjustments to the login React component. IDK whether that should hold up this PR.

Commit:
Close all menus before opening selected one.

- Renamed SignInProfileButton to SearchHelpSignIn.
- Added SignInProfileMenu.jsx

melwoodard commented:
I don't think it needs to, will split it out to a separate issue.

jbalboni committed:
Add school selection page to 1990e (#5059)

cvalarida committed:
5490 previous benefits page (#5058)

* WIP

* Previous benefits take 1

* Move 'other' field

* Expanding fields bug fix

* Add noPreviousBenefits option

* Remove the obnoxious claimSponsorService

With @jbalboni's fix, I can use expandUnder like it was meant to be used. Woot!

* Update unit test

* Required fields fix

* Remove comma

* Fix checkbox widget

* Update unit test

* Prefer definitions from vets-json-schema

* Added definitions to unit test

annekainicUSDS committed:
Make additional fixes

cvalarida committed:
Make sure the button is a button (#5069)

It was defaulting to submit the form, so making sure type="button" ensures that it has no default behavior (i.e. submitting the form).

annekainicUSDS committed:
1990e Chapter 5 Employment History (#5073)

jbalboni committed:
Use right address field on 1990e and 5490 (#5075)

cvalarida committed:
Edu applicant information page (#5065)

* WIP

* Pull out applicantInformation to pages

* Use direct deposit page again

* Modify applicantInformation()

* Capitalize the title

* Fix typo and use schema param

* Update unit test

* Use only schema from params

robbiethegeek-usds committed:
Adding the updates from @hollomancer (#5078)

Thanks to @hollomancer for your PR, moving over so I can merge since this blocked by our checks on your repo.

aub committed:
Set the environment as a tagmanager variable

We shouldn't need to pass the environment with every event we record in
tagmanager because it's a variable and so only needs to be set once.
This follows Google's suggestions about how to preset such a variable.
In addition, it stops loading GA at all in development mode.

Commit:
1990e education history page and unit tests (#5060)

Commit:
Fix label-rendering in review pages (#5071)

jbalboni committed:
5490 education history page (#5066)

Commit:
Updates model for test.

cvalarida committed:
Expound on the view: fields (#5074)

* Expound on the view: fields

Gives another example of what the `view:` fields can be used for.

* Add cookbook recipe

I'm not sure if it's much clearer this way, but it does give a more detailed example.

* Add conclusive statement about the back end

jbalboni committed:
Remove unnecessary sponsorVeteran object from 1990e (#5081)

jbalboni committed:
Add schema tests for 5490 (#5077)

Commit:
Add basic test for SearchHelpSignIn, SearchMenu, DropDown, HelpMenu

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

U-DON committed:
GIBCT Calculator Form (#5080)

* wip * calculator

* wip * calculator

* wip * calculator * housingAllow

* wip * lint-forced refactoring

* wip * calculator inputs, reducer, selector

* enhancements to profile page

* remove unused If import

* wip * calculator

* Typo.

* Separated subcomponents from render method.

* Get calculator to display.

* Fixed eligibility form case bug.

* Display correct inputs in calculator based on eligibility.

* Removed unnecessary input selector and keys.

* Display old GI Bill enrolled input. Also avoid processing dependents dropdown JSX if unnecessary.

* Added placeholder for tuition assist input. Refactored kicker amount input. Added missing state keys.

* Fixed incorrect prop.

* In progress refactoring and fixing of derived values calculations.

* Added getCurrency helper.

* Fixed lint issues and refactored some ad hoc components into simple conditionals.

* Fixed lint issues in calculator selector.

* Fixed errors.

* Pre-fill certain calculator fields after profile loads. Store dollar inputs as plain numbers and always display with dollar sign.

* Added in-state radio input and handling of changing the input to affect other calculator inputs.

* Added tuition assist input.

* Use isFinite from Lodash instead of isNumeric helper. Handle edge cases for value changes in dollar amount inputs.

* Fixed lint error.

* Added modal for tuition assistance.

* Wrapped switch cases in braces to allow declaration.

jbalboni committed:
Update confirmation page for 5490 and 1990e (#5085)

aub committed:
Add a target for the linter to only work on changed files (#5089)

* add a target for the linter to only work on changed files

* only lint changed files in pre-commit

* Fix link code

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

tiffanybbrown-va commented:
When I run `npm run test` locally, I get a bunch of errors for Edu components and 5490 schema. Not sure whether that's what's holding up these tests.

Commit:
Merge branch 'department-of-veterans-affairs/vets.gov-team-881' of github.com-tiffanybbrown-va:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/vets.gov-team-881

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

Commit:
Now, freshly linted

tiffanybbrown-va commented:
@melwoodard Ok: ready for re-review.

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

melwoodard reviewed (commented):
@tiffanybbrown-va the nav changes look great! Design approved on that front, though I do see the same homepage layout issues I noted in #5103.

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

jcmeloni-usds commented:
@tiffanybbrown-va @melwoodard This looks to be super close to being able to merge -- is that correct?

melwoodard commented:
The nav looks great, but there are some layout issues on the homepage I noted above in my review. I'm looking at it locally, but you can also see it at https://vetsgov-pr-5103.herokuapp.com/

Commit:
Merge branch 'master' into department-of-veterans-affairs/vets.gov-team-881

Commit:
Fixing visual regressions in DropDown Menu pattern.

Commit:
Fixing visual regression for new navigation menu.

Commit:
Added optional title property to IconSearch.

Icon's <title> text is spoken by screen readers. Removed it so that veterans wouldn't hear
"search search" when navigating header.

tiffanybbrown-va commented:
@melwoodard @jcmeloni-usds It's ready for a re-check from a design POV, and for a code review.

jcmeloni-usds commented:
Thank you! Would be awesome to close this out today.  I'm cool if @melwoodard is cool.

melwoodard reviewed (approved):
LGTM!

jcmeloni-usds commented:
LGTM.  I'm pressing the big green button. :tada:

U-DON commented on src/js/login/reducers/login.js:
```diff
@@ -1,15 +1,31 @@
+  let menu;
+
+  for (menu in menus) {
+    if (menus.hasOwnProperty(menu)) {
```
Could avoid checking for this by iterating with `Object.keys(menus).forEach(...)`.

But since this is to overwrite the state, if we want to keep things "immutable", it's probably better to just explicitly set it with `_.set('utilitiesMenuIsOpen', initialState.utilitiesMenuIsOpen, menuState);`.

Or to apply it directly to the case below, it could succinctly be refactored as `_.set('utilitiesMenuIsOpen', { ...initialState.utilitiesMenuIsOpen, [action.menu]: action.isOpen }, state);`.

U-DON commented on src/sass/base/_va.scss:
```diff
@@ -244,8 +253,7 @@ a[href^=http] {
   height: 40px;
-  overflow: hidden;
-  line-height: 1em;
+  //padding-left: .5rem;
```
Should this comment just be deleted instead?

U-DON commented on src/sass/modules/_m-vet-nav.scss:
```diff
@@ -149,7 +141,7 @@ body.va-pos-fixed {
   a.vetnav-level1 {
     border-bottom: 1px solid transparent;
-
+    
```
Whitespace.

U-DON commented on test/login/components/SearchHelpSignIn.unit.spec.jsx:
```diff
@@ -0,0 +1,45 @@
+        help: false,
+        search: false
+      }
+    };
```
Could be written as `const signedInData = { ...loginData, currentlyLoggedIn: true };` to be explicit that the only difference here is changing the one prop to `true`.

U-DON commented on src/js/login/components/SearchHelpSignIn.jsx:
```diff
@@ -0,0 +1,77 @@
+      dispatch(toggleSearchHelpUserMenu(menu, isOpen));
+    }
+  };
+};
```
Since the args are the same, could just go with the shorthand of making `mapDispatchToProps` an object.

U-DON reviewed (commented):
LGTM overall. Just had some minor comments.

ayaleloehr commented:
@tiffanybbrown-va I want to make sure @U-DON 's comments here aren't lost since they were posted post-merge. Are they on your radar?
